weaver/processes/wps_package.py

```diff
@@ -396,6 +404,63 @@ def _update_package_compatibility(package):
     return package
 
 
+def _load_supported_schemas():
+    """
+    Loads :term:`CWL` schemas supported by `Weaver` to avoid validation errors when provided in requirements.
+
+    Use a similar strategy as :func:`cwltool.main.setup_schema`, but skipping the :term:`CLI` context and limiting
+    loaded schema definitions to those that `Weaver` allows. Drops extensions that could cause misbehaving
+    functionalities when other :term:`Process` types than :term:`CWL`-based :term:`Application Package` are used.
+
+    This operation must be called before the :class:`CWLFactory` attempts loading and validating a :term:`CWL` document.
+    """
+    schema_supported = [name.rsplit(":", 1)[-1] for name in CWL_REQUIREMENTS_SUPPORTED]
+
+    # explicitly omit dev versions, only released versions allowed
+    extension_resources = {
+        "v1.0": "extensions.yml",
+        "v1.1": "extensions-v1.1.yml",
+        "v1.2": "extensions-v1.2.yml",
+    }
+    for version, ext_version_file in extension_resources.items():
+        # use our own cache on top of cwltool cache to distinguish between 'v1.x' names
+        # pointing at "CWL standard", "cwltool-flavored extensions" or "weaver-flavored extensions"
+        if version in PACKAGE_SCHEMA_CACHE:
+            LOGGER.debug("Reusing cached CWL %s schema extensions.", version)
+            continue
+        LOGGER.debug("Loading CWL %s schema extensions...", version)
+        with open_module_resource_file(cwltool, ext_version_file) as r_file:
+            schema = yaml.safe_load(r_file)
+
+        extensions = schema["$graph"]
+        extensions_supported = []
+        extensions_imports = []
+        extensions_enabled = set()
+        extensions_dropped = set()
+        for ext in extensions:
+            if "name" not in ext and "$import" in ext:
+                extensions_imports.append(ext)
+                continue
+            ext_name = ext["name"]
```

Comment:
I don't think so, otherwise there would not be any name to register the class. If it does happen, it would be only when updating `cwltool`, and it should be detected right away.